controller/src/main/java/io/pravega/controller/server/rpc/grpc/v1/ControllerServiceImpl.java

```diff
@@ -372,32 +379,33 @@ public boolean checkAuthorization(String resource, AuthHandler.Permissions expec
                 allowedLevel = currentInterceptor.authorize(resource);
             }
             if (allowedLevel.ordinal() < expectedLevel.ordinal()) {
-                return false;
+                throw new AuthorizationException("Access not allowed");
             }
-            return true;
-        } else {
-            return true;
         }
+        return "";
     }
 
-    public boolean checkAuthorizationWithToken(String resource, AuthHandler.Permissions expectedLevel) {
+    private String checkAuthorizationWithToken(String resource, AuthHandler.Permissions expectedLevel) throws AuthorizationException {
```

Comment:
This method name is strange. It is called 'checkAuthorizationWithToken' but it's actually creating a token. Can you clarify the name?

Reply:
Renamed to `checkAuthorizationAndCreateToken`. Do let me know whether this makes sense.